src/transformers/models/blip/modeling_blip.py

```diff
@@ -1252,6 +1252,14 @@ def __init__(self, config: BlipConfig):
         # Initialize weights and apply final processing
         self.post_init()
 
+    @classmethod
+    def can_generate(cls) -> bool:
```

Comment:
In this case we can make `BlipForQuestionAnswering` inherit `GenerationMixin`. This is my bad, we don't have any place in the docs saying when we should inherit `GenerationMixin`.

In a nutshell, we want `GenerationMixin` when we want the model to be able to call `GenerationMixin.generate` or if their custom `generate` calls `GenerationMixin.generate`.

When a model inherits `GenerationMixin`, it triggers a few things across our codebase, including:
- the model will load a Generation Config
- some custom pipeline behavior
- generation tests are run